src/Evolution/DgSubcell/Reconstruction.cpp

```diff
@@ -36,9 +37,7 @@ void reconstruct(const gsl::not_null<DataVector*> dg_u,
                  const DataVector& subcell_u_times_projected_det_jac,
                  const Mesh<Dim>& dg_mesh,
                  const Index<Dim>& subcell_extents) noexcept {
-  if (UNLIKELY(dg_u->size() != dg_mesh.number_of_grid_points())) {
-    *dg_u = DataVector(dg_mesh.number_of_grid_points());
-  }
+  destructive_resize_components(dg_u, dg_mesh.number_of_grid_points());
```

Comment:
Actually the documenation for `destructive_resize_components` has a warning not to use it on DataVectors as it claims no resize is done in that case. So either the documentation is wrong, or this is not resizing...   If 
 this function should not be used on a DataVector, perhaps it needs to be caught at compile time... 
 @moxcodes?

Reply:
I agree on both counts - the `destructive_resize_components` iterates over elements and resizes each, so performs no operation in the case of a `VectorImpl`. I also agree that I should add a static assert so that this is safer.